app/src/gui/components/alert.tsx

```diff
@@ -47,54 +36,64 @@ const useStyles = createUseStyles({
   },
 });
 
+/**
+ * SystemAlert component is responsible for rendering alerts in the form of snackbars at the bottom-center of the screen.
+ * Alerts are shown one at a time, with different durations based on their severity level.
+ * 
+ * @component
+ * @example
+ * return (
+ *   <SystemAlert />
+ * )
+ */
+
 export default function SystemAlert() {
   const classes = useStyles();
   const globalState = useContext(store);
-  const {dispatch} = globalState;
+  const { dispatch } = globalState;
   const alerts = globalState.state.alerts;
 
-  if (alerts.length === 0) return <></>;
-
-  const oldest_alert = alerts[alerts.length - 1]; // adjust the sequence to display the latest alert on the top, instead of bottom
-
+  /**
+  * Handles the closing of the current alert.
+  *
+  * @param {string} key - The unique key identifying the alert to be closed.
+  */
   const handleClose = (key: string) => {
     dispatch({
       type: ActionType.DELETE_ALERT,
-      payload: {
-        key: key,
-      },
+      payload: { key },
     });
   };
-  // this code is to move the successful message after 2 second, need to be updated in the next stage
-  setTimeout(() => {
-    if (
-      oldest_alert !== undefined &&
-      ['success', 'info'].includes(oldest_alert.severity)
-    )
-      handleClose(oldest_alert.key);
-  }, 3000);
-  // if (alerts.length > 0) console.log(oldest_alert.severity);
+
+  const currentAlert = alerts.length > 0 ? alerts[alerts.length - 1] : null;
+
   return (
     <ThemeProvider theme={theme}>
       <div className={classes.root}>
-        {alerts.length > 0 ? (
+        {currentAlert && (
```

Comment:
I know you didn't write this code but rather than conditionally rendering the `Snackbar` component you can move this logic into the `open` parameter. This might help with your animations.

Reply:
Gotcha! 
Code Updated.